bench/bin/throughput.rs

```diff
@@ -92,6 +99,190 @@ fn start_ingester(
+                    status
+                ),
+                Err(err) => panic!(
+                    "Failed to make HTTP request to metrics endpoint - reason: {:?}",
+                    err
+                ),
+            }
+        };
+
+        state.pop().map(|metric| (metric, state))
+    })
+}
+
+#[allow(dead_code)]
+pub struct MetricsRecorder {
+    server: tokio::task::JoinHandle<Vec<Sample>>,
+    abort_handle: AbortHandle,
+}
+
+impl MetricsRecorder {
+    #[allow(dead_code)]
+    pub fn start(port: u16, scrape_delay: Option<Duration>) -> MetricsRecorder {
+        let (abort_handle, abort_registration) = AbortHandle::new_pair();
+        MetricsRecorder {
+            server: tokio::spawn({
+                async move {
+                    Abortable::new(stream_agent_metrics(port, scrape_delay), abort_registration)
+                        .collect::<Vec<Sample>>()
+                        .await
+                }
+            }),
+            abort_handle,
+        }
+    }
+
+    #[allow(dead_code)]
+    pub async fn stop(self) -> Vec<Sample> {
+        self.abort_handle.abort();
+        match self.server.await {
+            Ok(data) => data,
+            Err(e) => panic!("error waiting for thread: {}", e),
+        }
+    }
+}
+
+fn data_pair_for(name: &str) -> impl Fn(&Sample) -> Option<(i64, f64)> + '_ {
+    move |s: &Sample| match (&s.value, &s.labels.get("outcome")) {
+        (Value::Untyped(raw), Some("success")) if s.metric.as_str() == name => {
+            Some((s.timestamp.timestamp_millis(), *raw))
+        }
+        _ => None,
+    }
+}
+// **** IMPORTED **************************
+
+// Sample number of lines
+fn calculate_fs_line_metrics(samples: &[Sample]) -> (f64, f64) {
+    let fs_sample_data = samples
+        .iter()
+        .filter_map(|s| match s.value {
+            Value::Counter(raw) if s.metric.as_str() == "logdna_agent_fs_lines" => Some(raw),
+            _ => None,
+        })
+        .zip(samples.iter().filter_map(|t| match t.value {
+            Value::Counter(_) if t.metric.as_str() == "logdna_agent_fs_lines" => {
+                Some(t.timestamp.timestamp_millis())
+            }
+            _ => None,
+        }))
+        .collect::<Vec<(f64, i64)>>();
+
+    let fs_total_time = (fs_sample_data.last().unwrap().1 - fs_sample_data[0].1) / 1000;
+    let fs_total_lines = fs_sample_data.last().unwrap().0;
+    let fs_lines_rate = fs_total_lines / fs_total_time as f64;
+
+    (fs_total_lines, fs_lines_rate)
+}
+
+// Sample maximum memory
+fn calculate_memory_max(samples: &[Sample]) -> f64 {
+    let mut max_value: f64 = 0.0;
+    let private_virtual_memory = samples
+        .iter()
+        .filter_map(|m| match m.value {
+            Value::Gauge(raw) if m.metric.as_str() == "process_virtual_memory_bytes" => Some(raw),
+            _ => None,
+        })
+        .collect::<Vec<f64>>();
+
+        for &val in private_virtual_memory.iter() {
```

Comment:
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.max

Reply:
Implemented the max method. It apparently doesn't handle `f34` or `f64` well but, thankfully, there was a workaround listed in the docs.